auth_jwt/tests/test_auth_jwt.py

```diff
@@ -0,0 +1,406 @@
+            # first validator rejects the token because of invalid audience
+            with self.assertRaises(UnauthorizedInvalidToken):
+                self.env["ir.http"]._auth_method_jwt_validator2()
+            # second validator accepts the token
+            self.env["ir.http"]._auth_method_jwt_validator3()
+
+    def test_auth_method_valid_token_two_validators_one_bad_audience_chained(self):
+        validator2 = self._create_validator("validator2", audience="bad")
+        validator3 = self._create_validator("validator3")
+
+        validator2.next_validator_id = validator3
+        authorization = "Bearer " + self._create_token()
+        with self._mock_request(authorization=authorization):
+            self.env["ir.http"]._auth_method_jwt_validator2()
+
+    def test_auth_method_invalid_token(self):
+        # Test invalid token via _auth_method_jwt
+        # Other types of invalid tokens are unit tested elswhere.
+        self._create_validator("validator4")
+        authorization = "Bearer " + self._create_token(audience="bad")
+        with self._mock_request(authorization=authorization):
+            with self.assertRaises(UnauthorizedInvalidToken):
+                self.env["ir.http"]._auth_method_jwt_validator4()
+
+    def test_auth_method_invalid_token_on_chain(self):
+        validator1 = self._create_validator("validator", issuer="http://other.issuer")
+        validator2 = self._create_validator("validator2", audience="bad audience")
+        validator3 = self._create_validator("validator3", secret_key="bad key")
+        validator4 = self._create_validator(
+            "validator4", issuer="http://other.issuer", audience="bad audience"
+        )
+        validator5 = self._create_validator(
+            "validator5", issuer="http://other.issuer", secret_key="bad key"
+        )
+        validator6 = self._create_validator(
+            "validator6", audience="bad audience", secret_key="bad key"
+        )
+        validator7 = self._create_validator(
+            "validator7",
+            issuer="http://other.issuer",
+            audience="bad audience",
+            secret_key="bad key",
+        )
+        validator1.next_validator_id = validator2
+        validator2.next_validator_id = validator3
+        validator3.next_validator_id = validator4
+        validator4.next_validator_id = validator5
+        validator5.next_validator_id = validator6
+        validator6.next_validator_id = validator7
+
+        authorization = "Bearer " + self._create_token()
+        with self._mock_request(authorization=authorization):
+            with self.assertRaises(UnauthorizedCompositeJwtError) as composite_error:
+                self.env["ir.http"]._auth_method_jwt_validator()
+            self.assertEqual(
+                str(composite_error.exception),
+                "401 Unauthorized: "
+                + "Multiple errors occurred during JWT chain validation:\n"
```

Comment:
```suggestion
                "Multiple errors occurred during JWT chain validation:\n"
```

Was this `+` added automatically by ruff ?